tests/integration/ha_tests/application-charm/src/charm.py

```diff
@@ -72,12 +88,23 @@ def _on_start(self, _) -> None:
         """Only sets an Active status."""
         self.unit.status = ActiveStatus()
 
-    def _on_database_created(self, _) -> None:
-        """Event triggered when a database was created for this application."""
-        self._start_continuous_writes(1)
-
     def _on_endpoints_changed(self, _) -> None:
         """Event triggered when the read/write endpoints of the database change."""
+        if self._connection_string is None:
+            return
+
+        if not self.app_peer_data.get(PROC_PID_KEY):
+            return None
+
+        with open(CONFIG_FILE, "w") as fd:
+            fd.write(self._connection_string)
+            os.fsync(fd)
+
+        try:
+            os.kill(int(self.app_peer_data[PROC_PID_KEY]), signal.SIGKILL)
```

Comment:
Tried SIGHUPing the script, however `psycopg2` hangs the session if connection is lost during execution (freeze postgesql process test) and I was unable to figure out the configuration necessary to make it timeout. Haven't tried with `psycopg3`, it might be worthwhile to investigate.